lib/rdf/model/uri.rb

```diff
@@ -739,7 +739,7 @@ def ==(other)
         # If other is a Literal, reverse test to consolodate complex type checking logic
         other == self
       when String then to_s == other
-      when URI then hash == other.hash && to_s == other.to_s
+      when URI then hash == other.hash
```

Comment:
It's possible (but unlikely) for the hashes of to URIs to be the same while the values are actually different. This just tries to short-cut the check by first checking hash values to exit early.